### DISCUSSION TITLE
Welcome to Nucleoid Discussions!

### DISCUSSION THREAD

**Discussion by canmingir** [NucleoidAI/Nucleoid#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jacksonkasi0** [comment]
**Nucleoid JS** is really awesome 🤩

**Comment by mtmgit** [comment]
We have numerous interns in addition to myself that want to start the process of building a core application that uses declarative programming and to understand Nucleoid capabilities and app performance outcomes of sprint deliverables. This means first understanding and creating the application logic which could be handled using your DSL.  This is where we need to hone our skills. 

I believe being a "newbie" developer myself that instructing a program on "what needs to be done" versus "how to do it" is very counterintuitive for many interns with the low-level imperative fundamentals that are being taught in schools. Many necessary - but the complexity of "knowing everything" seems almost baked into the way student think, but also may be chasing them out of computer science. Maybe related to perceived "self-value" through unneeded complexity. 

Potentially much time would be better spent where there is currently little focus: The logic of the overall application, given the underlying DSL. Logic at its deepest level is needed to lock down core aspects of our program like the foundation of a house with the structural integrity beyond reproach. 

After reading about the unfolding structure in UNIX and the checklists below, I felt compelled explore a  “Nucleoid core” to obfuscate underlying abstractions within our application build. 

Read these questions that are definitely us!

1) constantly working around the tool rather than with the tool.

2) struggling to solve a straightforward problem that squarely belongs in the domain of the tool you are using

3) longing to ditch the tool and do the whole thing yourself inside a _ for- loop_.
